bucket.go

```diff
@@ -371,9 +371,9 @@ func (b *Bucket) NextSequence() (uint64, error) {
 
 // ForEach executes a function for each key/value pair in a bucket.
 // Because ForEach uses a Cursor, the iteration over keys is in lexicographical order.
-// If the provided function returns an error then the iteration is stopped and
-// the error is returned to the caller. The provided function must not modify
-// the bucket; this will result in undefined behavior.
+// If the provided function returns an error then the iteration is stopped and the error is returned to the caller.
+// If the value is nil, then it's a nested bucket.
+// The provided function must not modify the bucket; this will result in undefined behavior.
```

Comment:
How about this?  The key point is to put "`If the value is nil,....`" right after the first sentence, so that users are still in the context of "`each key/value pair...`".

```
// ForEach executes a function for each key/value pair in a bucket.
// If the value is nil, then it's a bucket, and the key is the bucket name.
//
// Because ForEach uses a Cursor, the iteration over keys is in lexicographical
// order. If the provided function returns an error then the iteration is stopped
// and the error is returned to the caller. The provided function must not modify
// the bucket; this will result in undefined behavior.
```

Reply:
Yes, better. I also added "nested" word to emphasize it's about bucket-in-bucket scenario.

Reply:
> I also added "nested" word to emphasize it's about bucket-in-bucket scenario.

I am OK to add "nested". But the key reason isn't bucket-in-bucket; instead, it's `store a bucket in a key to create nested buckets`.

Although there is a implicit root bucket, it's transparent to users. From users perspective, they can create top level buckets (similar to tables in traditional RDBMS), and also child buckets under any bucket.

Reply:
https://github.com/etcd-io/bbolt/#nested-buckets